nodes/list_masks/mask_join.py

```diff
@@ -87,28 +103,29 @@ def apply_mask(self, mask, data_t, data_f):
         return out
 
     def get_level(self, mask, data_t, data_f, level):
+        if self.choice:
+            apply_mask = self.apply_choice_mask
+        else:
+            apply_mask = self.apply_mask
+            
         if level == 1:
             out = []
-            param = (mask, data_t, data_f)
+            param = list_match_func[self.list_match_global]([mask, data_t, data_f])
             if not all((isinstance(p, (list, tuple)) for p in param)):
                 print("Fail")
                 return
-            max_index = min(map(len, param))
-            if self.choice:
-                apply_mask = self.apply_choice_mask
-            else:
-                apply_mask = self.apply_mask
 
-            for i in range(max_index):
-                out.append(apply_mask(mask[i], data_t[i], data_f[i]))
+            for m, t, f in zip(*param):
+                out.append(apply_mask(m, t, f))
             return out
-        elif level > 2:
+        elif level >= 2:
```

Comment:
It seems also requires next version of the node. I doubt that else statement did the same for level 2.

Reply:
Well, yeah, although it is broken, if people created node tree based on broken node, it won't behave the same, they would need to change the level to apply mask. If I create a new node, the old one will be moved to deprecated, is that it?

Reply:
A node becomes and old one when its code gets into the old nodes folder.
http://nortikin.github.io/sverchok/docs/contributing/node_api.html#create-new-node-version
No need to create new node, just change version of this one and copy old code in the old nodes folder.